hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -412,6 +412,40 @@ private void validateTableIsIcebergTableOrView(
     }
   }
 
+  /**
+   * Check whether table or metadata table exists.
+   *
+   * @param identifier a table identifier
+   * @return true if the table exists, false otherwise
```

Comment:
nit, should we note the behavior for when there's a conflicting hive table?

Reply:
I think for the HiveCatalog implementation that comment would in fact add some extra value for the users so +1